File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/domain/usecase/GetKeyboardTypeUseCase.java
package space.lopatkin.spb.helpboardgamecard.domain.usecase;

import space.lopatkin.spb.helpboardgamecard.domain.repository.AppRepository;
import space.lopatkin.spb.helpboardgamecard.domain.model.KeyboardType;

public class GetKeyboardTypeUseCase {

    private AppRepository repository;

    public GetKeyboardTypeUseCase(AppRepository repository) {
        this.repository = repository;
    }

    public KeyboardType execute() {
        return KeyboardType.getOrdinalFrom(repository.getKeyboardType());
    }

}

File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/domain/usecase/GetKeyboardTypeUseCase.kt
package space.lopatkin.spb.helpboardgamecard.domain.usecase

import kotlinx.coroutines.flow.Flow
import space.lopatkin.spb.helpboardgamecard.domain.model.KeyboardType
import space.lopatkin.spb.helpboardgamecard.domain.repository.SettingsRepository

class GetKeyboardTypeUseCase(private val repository: SettingsRepository) {

    fun execute(): Flow<KeyboardType> {
        return repository.getKeyboardType()
    }

}
